server/src/main/java/keywhiz/auth/UserAuthenticatorFactory.java

```diff
@@ -13,5 +13,5 @@
   /**
    * Builds an authenticator from username/password credentials to a {@link User}.
    */
-  Authenticator<BasicCredentials, User> build(DBI dbi);
+  Authenticator<BasicCredentials, User> build(UserDAO userDAO);
```

Comment:
My thinking original is that `UserAuthenticatorFactory` should support any sort of user authenticator. Passing a `DBI` (or `DSLContext`) lets it use any table, including one not in keywhiz-model. This is speculative though, trying to build for the future.